src/tools/rbd_mirror/image_map/Policy.h

```diff
@@ -0,0 +1,94 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_RBD_MIRROR_IMAGE_MAP_POLICY_H
+#define CEPH_RBD_MIRROR_IMAGE_MAP_POLICY_H
+
+#include <tuple>
+#include <vector>
+
+#include "Types.h"
+#include "common/RWLock.h"
+#include "include/Context.h"
+#include "tools/rbd_mirror/types.h"
+#include "common/AsyncOpTracker.h"
+#include "include/rados/librados.hpp"
+
+namespace librbd { class ImageCtx; }
+
+namespace rbd {
+namespace mirror {
+namespace image_map {
+
+template <typename ImageCtxT = librbd::ImageCtx>
+class Policy {
+public:
+  Policy(librados::IoCtx &ioctx);
+  virtual ~Policy() {
+  }
+
+  // interfaces for anyone using the map
+
+  // init (load) the instance map from disk
+  void init(Context *on_finish);
+
+  // shut down map operations
+  void shut_down(Context *on_finish);
+
+  // handle additional/removal of images
+  void update_images(const std::string &mirror_uuid,
+                     rbd::mirror::ImageIds &&added_image_ids,
+                     rbd::mirror::ImageIds &&removed_image_ids,
+                     MapSpecRef mapspec, Context *on_finish);
+
```

Comment:
Need interface for handling the ACK from instances that the image was added / removed as demanded.